### ISSUE TITLE
CA1062 still fires when ValidatedNotNullAttribute is present in a specific case

### ISSUE THREAD

**Issue by Julie-BC** [dotnet/roslyn-analyzers#3177]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.6

#### Diagnostic ID

[CA1062](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062?view=vs-2019)

#### Repro steps

1. Apply a parameter validation with a ValidatedNotNullAttribute in the beginning of a public method
2. Call some other functions : one intermediary method which calls a delegate function
3. Use the parameter reference 

#### Expected behavior  
Every usage of parameter reference is possible without CA1062 rule violation in the method (after validation)  
#### Actual behavior  
After the intermediary method call, validation seems forgotten, and CA1062 is violated for all the next references.   

If the IntermediaryMethod doesn't contain a function the attribute behaves as expected. 

```csharp
using System;

namespace IssueCA1062
{
    public class ExampleClass
    {
        private Func<object, object> Function { get; }

        public ExampleClass()
        {
            Function = (object test) => { return test; };
        }

        public void IntermediaryMethod(object param)
        {
            var result = Function(param);
        }
        
        public void PublicMethodWithParam(string str)
        {
            ArgsValidation.ThrowIfNull(str, nameof(str));
            var firstAccess = str.Length; // This access is ok, no CA1062 violation thanks to previous validation

            IntermediaryMethod(new object());

            var nextAccess = str.Length; // This access is not ok, CA1062 violation because of previous method calling tree
        }
    }

    [AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
    public sealed class ValidatedNotNullAttribute : Attribute
    {
    }
    
    public static class ArgsValidation
    {
        public static void ThrowIfNull([ValidatedNotNull] object argument, string argumentName)
        {
            if (argument == null)
            {
                throw new ArgumentNullException(nameof(argumentName));
            };
        }
    }
}  
```

**Comment by Evangelink** [comment]
@Julie-BC The behavior is correct with the latest version. Could you please update your package?

**Comment by mavasani** [comment]
Thanks @Evangelink. Marking this issue as closed. @Julie-BC please let us know it is repros with latest package.

**Comment by Julie-BC** [comment]
@Evangelink , @mavasani : I'm sorry but this is still not working with the version 2.9.8 and the same code example

![image](https://user-images.githubusercontent.com/59656185/72437338-cfdfc600-37a2-11ea-9364-e54bb095dfdf.png)

**Comment by Evangelink** [comment]
That's interesting, I do manage to reproduce the error while I couldn't last time. I'll have a look and see if I manage to find the problem.

**Comment by Evangelink** [comment]
@Julie-BC Could you reopen the issue?

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/2919
